src/OrchardCore.Modules/OrchardCore.Templates/Services/AdminTemplatesShapeBindingResolver.cs

```diff
@@ -65,6 +68,36 @@ public async Task<ShapeBinding> GetShapeBindingAsync(string shapeType)
             }
         }
 
+        public async Task<IEnumerable<string>> GetShapeBindingNamesAsync(Func<string, bool> predicate, bool adminTemplate)
+        {
+            if (!adminTemplate)
+            {
+                return null;
+            }
+
+            if (_templatesDocument == null)
+            {
+                _templatesDocument = await _templatesManager.GetTemplatesDocumentAsync();
+            }
+
+            var query = _templatesDocument.Templates
+                .Where(t => predicate(t.Key))
+                .Select(s => s.Key);
+
+            var localTemplates = _previewTemplatesProvider.GetTemplates();
+
+            if (localTemplates != null)
+            {
+                query = query.Union(
```

Comment:
Union or Append ?

Reply:
Creating unique list of shape names without duplicates